sdk/python/feast/infra/online_stores/remote.py

```diff
@@ -0,0 +1,181 @@
+                    response_json["metadata"]["feature_names"]
+                ):
+                    if self._check_if_feature_requested(
+                        feature_name, requested_features
+                    ):
+                        is_present = (
+                            True
+                            if response_json["results"][index]["statuses"][
+                                feature_value_index
+                            ]
+                            == "PRESENT"
+                            else False
+                        )
+                        if is_present:
+                            message = python_values_to_proto_values(
+                                [
+                                    response_json["results"][index]["values"][
+                                        feature_value_index
+                                    ]
+                                ],
+                                ValueType.UNKNOWN,
+                            )
+                            feature_values_dict[feature_name] = message[0]
+                        else:
+                            feature_values_dict[feature_name] = ValueProto()
+
+                result_tuples.append((event_ts, feature_values_dict))
+                feature_value_index += 1
+            return result_tuples
+        else:
+            error_msg = f"Unable to retrieve the online store data using feature server API. Error_code={response.status_code}, error_message={response.reason}"
+            logger.error(error_msg)
+            raise RuntimeError(error_msg)
+
+    def _construct_online_read_api_json_request(
+        self,
+        entity_keys: List[EntityKeyProto],
+        table: FeatureView,
+        requested_features: Optional[List[str]] = None,
+    ):
+        api_requested_features = []
+        if requested_features is not None:
+            for requested_feature in requested_features:
+                api_requested_features.append(f"{table.name}:{requested_feature}")
+
+        entity_values = []
+        entity_key = ""
+        for row in entity_keys:
+            entity_key = row.join_keys[0]
+            entity_values.append(
+                getattr(row.entity_values[0], row.entity_values[0].WhichOneof("val"))
+            )
+
+        req_body = json.dumps(
+            {
+                "features": api_requested_features,
+                "entities": {entity_key: entity_values},
+            }
+        )
+        return req_body
+
+    def _check_if_feature_requested(self, feature_name, requested_features):
+        for requested_feature in requested_features:
+            if feature_name in requested_feature:
+                return True
+        return False
+
+    def _get_event_ts(self, response_json) -> datetime:
+        event_ts = ""
+        for index, result in enumerate(response_json["results"]):
+            if index == 1:
+                event_ts = result["event_timestamps"][0]
```

Comment:
isn't this equivalent to the following?
```py
# maybe check len of first array
event_ts = response_json["results"][1]["event_timestamp"][0]
```

Reply:
just trying to avoid the array index exception, i can check the length of the array before instead of iterating over it.